docs/html/conf.py

```diff
@@ -1,16 +1,16 @@
 """Sphinx configuration file for pip's documentation."""
 
-import glob
 import os
 import pathlib
-import re
 import sys
-from typing import List, Tuple
 
-# Add the docs/ directory to sys.path, because pip_sphinxext.py is there.
+# Add the docs/ directory to sys.path to load the common config,
+# and pip_sphinxext.py
 docs_dir = os.path.dirname(os.path.dirname(__file__))
 sys.path.insert(0, docs_dir)
 
+from common_conf import copyright, project, release, version  # noqa: E402
```

Comment:
You can use `__all__` to signify the values are re-exported.

Reply:
> You can use `__all__` to signify the values are re-exported.

Are you suggesting add them to `__all__` in this file, instead of the assignments below? Or add these to `__all__` in `common_conf` and just `import *` here?

Reply:
Why not simply ignore the Ruff rule?

```suggestion
from common_conf import copyright, project, release, version  # noqa: E402, F401
```
---
> Are you suggesting add them to `__all__` in this file, instead of the assignments below?

Names declared in `__all__` become part of the module's interface and thus Ruff won't delete unused imports that populate those names. AFAIU, this is what @uranusjr was suggesting. I think this is even less obvious to a reader.

Alternatively we could do `from common_config import version as version` for each configuration option to declare them as re-exports.

```python
from common_conf import copyright as copyright  # noqa: E402
from common_conf import project as project  # noqa: E402
from common_conf import release as release  # noqa: E402
from common_conf import version as version  # noqa: E402
```

But this seems verbose IMO.

Reply:
> Why not simply ignore the Ruff rule?

No reason not to, see c04ff52c6d44309693a7aea17648e4a058969fe5